docs/modules/ROOT/pages/deploy_related/integration_nova.adoc

```diff
@@ -21,16 +21,288 @@ We would want to integrate with Nova to support the same set of user operations,
+dns_name:''}
+|
+|/v2.0/ports/ {port_id}
+|/project/{project_id}/ ports/{port_id}
+|stage 4: unbind instance port
+
+|update_port
+|put
+|port: {dns_name: ''}
+|
+|/v2.0/ports/ {port_id}
+|/project/{project_id}/ ports/{port_id}
+|stage 4: reset port dns name
+
+|update_port
+|put
+|port: {'binding:host_id': host, device_owner: 'compute:zone', 'binding:profile':{} }
+|
+|/v2.0/ports/ {port_id}
+|/project/{project_id}/ ports/{port_id}
+|stage 2: update port binding for instance in build network resource(have request port id and port exist in network)
+
+|update_port
+|put
+|port: {'binding:host_id': host, device_owner: 'compute:zone', 'binding:profile':{}, 'dns_name': network.dns_domain or instance.hostname, mac_address: mac}
+|
+|/v2.0/ports/ {port_id}
+|/project/{project_id}/ ports/{port_id}
+|stage 2: update port for instance in build network resource
+
+|delete_port
+|delete
+|
+|
+|/v2.0/ports/ {port_id}
+|/project/{project_id}/ ports/{port_id}
+|stage 3;delete port for instance
+
+|list_floatingips
+|get
+|fixed_ip_address: fixed_ip, port_id: port_id
+|
+|/v2.0/floatingips
+|
+|stage 2: get port floatingip if have request port and request port is exist
+
+|show_quota
+|get
+|
+|
+|/v2.0/quotas/ {project_id}
+|
+|stage 1: confirm tenant user have enough ports resources
+
+|list_extensions
+|get
+|
+|return example: {'extensions': {{'updated': "2017-07-17T10:00:00-00：00",
+name: port_binding_ extended,
+links: [],
+alias: binding-extended,
+description: "Expose port binding of a virtual port to external application"}}}
+|/v2.0/extensions
+|
+|stage 1, 2: get all support extension options
+
+|get_auto_allocated_ topology
+|get
+|
+|
+|/v2.0/auto-allocated-topology/ {project_id}
+| optional api
+| stage 2: auto allocate network if no request network and no available network
+
+|list_security_ groups
+|get
+|tenant_id: project_id
+|
+|/v2.0/security-groups
+|no support
+| stage 2: process security groups for instance in build network resource
+|====================
+
 === VM Creation Workflow
 image::vm_create.png["VM creation workflow", width=1024, link="vm_create.png"]
 
 == Required Changes
+https://github.com/openstack/python-neutronclient[neutronclient project]
+
+Nova use python-neutronclient to call neutron apis.Only need to change neutronclient/v2.0/client:Client class.
+
+== How nova client identify alcor server url
+In openstack, there are a auth server keystone, it can offer server url auth and endpoint catalog.  +
+https://docs.openstack.org/mitaka/cli-reference/keystone.html[keystone online docs]
+
+So alcor need register endpoint in keystone. +
+Register:
+
+$ openstack endpoint create --region RegionOne
+network public http://<alcor_ip>:<port>
+
+$ openstack endpoint create --region RegionOne
+network internal http://<alcor_ip>:<port>
+
+$ openstack endpoint create --region RegionOne
+network admin http://<alcor_ip>:<port>
+
+After register in keystone, nova can get alcor endpoint from keystone. No need to change nova config file.
```

Comment:
Besides the API endpoint registration, do we need to consider registering Alcor as one of OpenStack service and register a service user for Alcor for API authentication purpose. Following the current OpenStack service management approach? https://docs.openstack.org/keystone/latest/admin/manage-services.html

Reply:
This is a good point @cj-chung @Gzure . In addition to instruction to integrate with Nova, we also need the detailed instruction on integration with other OpenStack component. 

@Gzure we could either list it here or create a separate document. It is your call.